core/trino-main/src/main/java/io/trino/operator/output/PositionsAppender.java

```diff
@@ -11,37 +11,38 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 package io.trino.operator.output;
 
 import io.trino.spi.block.Block;
-import io.trino.spi.block.BlockBuilder;
-import io.trino.spi.type.Type;
+import io.trino.spi.block.BlockBuilderStatus;
 import it.unimi.dsi.fastutil.ints.IntArrayList;
 
-import static java.util.Objects.requireNonNull;
+import javax.annotation.Nullable;
 
 public interface PositionsAppender
 {
-    void appendTo(IntArrayList positions, Block source, BlockBuilder target);
-
-    class TypedPositionsAppender
-            implements PositionsAppender
-    {
-        private final Type type;
-
-        public TypedPositionsAppender(Type type)
-        {
-            this.type = requireNonNull(type, "type is null");
-        }
-
-        @Override
-        public void appendTo(IntArrayList positions, Block source, BlockBuilder target)
-        {
-            int[] positionArray = positions.elements();
-            for (int i = 0; i < positions.size(); i++) {
-                type.appendTo(source, positionArray[i], target);
-            }
-        }
-    }
+    void appendTo(IntArrayList positions, Block source);
```

Comment:
I was thinking about keeping old `PositionAppender` with `void appendTo(IntArrayList positions, Block source, BlockBuilder target);` rather than modifying `PositionsAppender` interface (that would limit number of changes). As it's now I don't see a value in separate `PositionsAppender` and `BlockTypeAwarePositionsAppender`